screencaptor/src/main/java/com/wealthfront/screencaptor/ScreenCaptor.kt

```diff
@@ -80,50 +81,82 @@ object ScreenCaptor {
 
-    val screenshotName = "${screenshotFilename.toLowerCase(ENGLISH)}_${device}_${SDK_INT}_$screenshotNameSuffix"
-    val screenshotFile = "$screenshotDirectory/$screenshotName.${screenshotFormat.extension}"
+    val bitmap = createCanvasBitmap(rootViews)
+    val canvas = Canvas(bitmap)
 
-    Log.d(SCREENSHOT, "Posting to main thread for '$screenshotFile'")
     mainHandler.post {
-      Log.d(SCREENSHOT, "Root view has height(${view.height}) and width(${view.width})")
-      if (view.width == 0 || view.height == 0) {
-        throw IllegalStateException("This view ($view) has no height or width. Is ${view.id} the currently displayed activity?")
+      Log.d(SCREENSHOT, "Posting to main thread for '$screenshotId'")
+
+      rootViews.forEach { rootView ->
+        renderViewToCanvas(
+          canvas,
+          rootView,
+          viewIdsToExclude,
+          viewModifiers,
+          viewDataProcessor,
+          viewMutators,
+          screenshotName
+        )
       }
 
-      Log.d(SCREENSHOT, "Mutating views as needed!")
-      ViewTreeMutator.Builder()
-        .viewMutators(viewMutators)
-        .mutateView(view)
-        .mutate()
-
-      Log.d(SCREENSHOT, "Disabling views: $viewIdsToExclude")
-      val initialStateOfViews = ViewVisibilityModifier.hideViews(view, viewIdsToExclude)
-      val initialDataOfViews = viewDataProcessor.modifyViews(view, viewModifiers)
-
-      view.post {
-        Log.d(SCREENSHOT, "Taking screenshot for '$screenshotFile'")
-        val bitmap = Bitmap.createBitmap(view.width, view.height, ARGB_8888)
-        val canvas = Canvas(bitmap)
-        view.draw(canvas)
-
-        if (!File(screenshotDirectory).exists()) {
-          Log.d(SCREENSHOT, "Creating directory $screenshotDirectory since it does not exist")
-          File(screenshotDirectory).mkdirs()
-        }
-
-        Log.d(SCREENSHOT, "Writing to disk for '$screenshotFile'")
-        bitmap.compress(screenshotFormat.compression, screenshotQuality.value, FileOutputStream(screenshotFile))
-        Log.d(SCREENSHOT, "Successfully wrote to disk for '$screenshotFile'")
-
-        Log.d(SCREENSHOT, "Enabling views: $viewIdsToExclude")
-        ViewVisibilityModifier.showViews(view, viewIdsToExclude, initialStateOfViews)
-        viewDataProcessor.resetViews(view, initialDataOfViews)
+      if (!File(screenshotDirectory).exists()) {
+        Log.d(SCREENSHOT, "Creating directory $screenshotDirectory since it does not exist")
+        File(screenshotDirectory).mkdirs()
       }
+
+      Log.d(SCREENSHOT, "Writing to disk for '$screenshotFilePath'")
+      bitmap.compress(
+        screenshotFormat.compression,
+        screenshotQuality.value,
+        FileOutputStream(screenshotFilePath)
+      )
+      Log.d(SCREENSHOT, "Successfully wrote to disk for '$screenshotFilePath'")
     }
   }
+
+  private fun createCanvasBitmap(rootViews: List<View>): Bitmap {
+    val canvasWidth = rootViews.map { it.width }.max()!!
```

Comment:
nit: there should be a maxBy function which does this, i think

Reply:
hah I actually had that at first. Since you think it's more intuitive I'll go back to that